src/sst/core/configGraph.cc

```diff
@@ -904,45 +922,42 @@ ConfigGraph::getCollapsedPartitionGraph()
     PartitionComponentMap_t& pcomps = graph->getComponentMap();
     PartitionLinkMap_t& plinks = graph->getLinkMap();
 
+
+    // Mark all Components as not visited
+    for ( ConfigComponentMap_t::iterator it = comps.begin(); it != comps.end(); ++it ) it->visited = false;
+
     // SparseVectorMap is slow for random inserts, so make sure we
     // insert both components and links in order of ID, which is the
     // key for the SparseVectorMap in both cases
     ComponentIdMap_t group;
     for ( ConfigComponentMap_t::iterator it = comps.begin(); it != comps.end(); ++it ) {
-        const ConfigComponent& comp = *it;
-
+        if ( it->visited ) continue;
         // Get the no-cut group for this component
         group.clear();
         getConnectedNoCutComps(it->id,group);
 
-
-        // Check to see if this has already been put in map.  Do this
-        // by seeing if the first item in the connected components is
-        // the current ID.  If not, then it's already in the list.
-        if ( *group.begin() == comp.id ) {
-            ComponentId_t id = pcomps.size();
-            pcomps.insert(PartitionComponent(id));
-            PartitionComponent& pcomp = pcomps[id];
-
-            // Iterate over the group and add the weights and add any
-            // links that connect outside the group
-            for ( ComponentIdMap_t::const_iterator i = group.begin(); i != group.end(); ++i ) {
-                const ConfigComponent& comp = comps[*i];
-                // Compute the new weight
-                pcomp.weight += comp.weight;
-                pcomp.group.insert(comp.id);
-
-                // Walk through all the links and insert the ones that connect
-                // outside the group
-                for ( LinkId_t id : comp.allLinks() ) {
-                    const ConfigLink& link = links[id];
-
-                    if ( !group.contains(COMPONENT_ID_MASK(link.component[0])) || !group.contains(COMPONENT_ID_MASK(link.component[1]) ) ) {
-                        pcomp.links.push_back(link.id);
-                    }
-                    else {
-                        deleted_links.insert(link.id);
-                    }
+        ComponentId_t id = pcomps.size();
+        pcomps.insert(PartitionComponent(id));
```

Comment:
We used to use the built-in map types, but they use a lot of extra memory, which for large graphs were sometimes causing us to run out of memory on rank 0.  Changing to a dense data structure fixed that problem.  It may not be an issue with the memory sizes of today, but it was in the past.  Making the change to return a reference would be trivial.

Reply:
@feldergast - the issue came up a lot running on Haswell (Mutrino) with large network simulations. I would still like to be able to use this platform for our larger runs if we can - i.e. I think your approach with the dense structures is still needed right now and I don't see that changing too much for another year or two.